pkg/liqonet/tunnel/wireguard/driver.go

```diff
@@ -184,15 +190,20 @@ func (w *Wireguard) ConnectToEndpoint(tep *netv1alpha1.TunnelEndpoint) (*netv1al
 	}
 
 	// delete or update old peers for ClusterID.
+	w.connectionsMutex.RLock()
 	oldCon, found := w.connections[tep.Spec.ClusterIdentity.ClusterID]
+	w.connectionsMutex.RUnlock()
 	if found {
 		// check if the peer configuration is updated.
 		if stringAllowedIPs == oldCon.PeerConfiguration[AllowedIPs] && remoteKey.String() == oldCon.PeerConfiguration[liqoconst.PublicKey] &&
 			endpoint.IP.String() == oldCon.PeerConfiguration[EndpointIP] && strconv.Itoa(endpoint.Port) == oldCon.PeerConfiguration[liqoconst.ListeningPort] {
 			// Update connection status.
-			return w.updateConnectionStatus(oldCon)
+			w.connectionsMutex.Lock()
+			w.connections[tep.Spec.ClusterIdentity.ClusterID] = tep.Status.Connection.DeepCopy()
+			oldCon = w.connections[tep.Spec.ClusterIdentity.ClusterID]
+			w.connectionsMutex.Unlock()
+			return oldCon, nil
```

Comment:
```suggestion
			return tep.Status.Connection.DeepCopy(), nil
```

Should be enough IIUC

Reply:
With the suggested solution the connection object inside the wireguard driver (w object) won't be updated

Reply:
I agree, but it seems to me that the latency info written there is never used (and the other values are equal nonetheless if you end up in that if).